### PR TITLE
reload node defs on cycle change

### PR DESCRIPTION
**Pull request by fefrain**

### CODE REVIEW

File: server/modules/survey/service/publish/jobs/nodeDefsValidationJob.js [hunk 1]
```diff
@@ -15,17 +15,20 @@ class NodeDefsValidationJob extends Job {
...
-      Survey.getNodeDefsValidation,
-      Validation.getFieldValidations,
-      R.forEachObjIndexed((nodeDefValidation, nodeDefUuid) => {
-        const nodeDef = Survey.getNodeDefByUuid(nodeDefUuid)(survey)
-        this.errors[NodeDef.getName(nodeDef)] = Validation.getFieldValidations(nodeDefValidation)
-      })
-    )(survey)
-
+    const survey = SurveyManager.fetchSurveyById(this.surveyId, true, false, tx)
+    const cycleKeys = R.pipe(Survey.getSurveyInfo, Survey.getCycles, R.keys)(survey)
```
[the remark this answers is not in the document]
Comment (fefrain) [reply server/modules/survey/service/publish/jobs/nodeDefsValidationJob.js]: ok

File: server/modules/nodeDef/api/nodeDefApi.js [hunk 2]
```diff
@@ -27,7 +30,8 @@ module.exports.init = app => {
 
       await NodeDefService.insertNodeDef(user, surveyId, nodeDefRequest)
 
-      await sendRespNodeDefs(res, surveyId)
+      const cycle = R.pipe(NodeDef.getCycles, R.head)(nodeDefRequest)
+      await sendRespNodeDefs(res, surveyId, cycle)
```
[the remark this answers is not in the document]
Comment (fefrain) [reply server/modules/nodeDef/api/nodeDefApi.js]: when you insert a nodeDef, you associate it always to one cycle

File: server/modules/collectImport/service/collectImport/dataImportJobs/recordsUniquenessValidationJob.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ class RecordsUniquenessValidationJob extends Job {
     this.total = 2
 
     // 1. fetch survey and node defs
-    this.survey = await SurveyManager.fetchSurveyAndNodeDefsAndRefDataBySurveyId(this.surveyId, true, true, false, false, this.tx)
+    this.survey = await SurveyManager.fetchSurveyAndNodeDefsAndRefDataBySurveyId(this.surveyId, null, true, true, false, false, this.tx)
```
[the remark this answers is not in the document]
Comment (fefrain) [reply server/modules/collectImport/service/collectImport/dataImportJobs/recordsUniquenessValidationJob.js]: it's something we will fix later on

[the remark this answers is not in the document]
Comment (minotogna) [reply server/modules/collectImport/service/collectImport/dataImportJobs/recordsUniquenessValidationJob.js]: not sure. records are always at cycle '0' in collectImport job..

**Pull request merged**